src/reporters/GriddedDataReporter.C

```diff
@@ -0,0 +1,39 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GriddedDataReporter.h"
+#include "DelimitedFileReader.h"
+#include "GriddedData.h"
+
+registerMooseObject("isopodApp", GriddedDataReporter);
+
+InputParameters
+GriddedDataReporter::validParams()
+{
+  InputParameters params = GeneralReporter::validParams();
+  params.addClassDescription(
+      "Reporter to hold griddedData parameter values and their spatial and temporal coordinates");
+  params.addRequiredParam<FileName>(
+      "parameter_file_name",
+      "text file for every parameter, includes the parameter values, its spatial and temporal "
+      "distribution");
+  return params;
+}
+
+GriddedDataReporter::GriddedDataReporter(const InputParameters & parameters)
+  : GeneralReporter(parameters),
+    _file_name(getParam<FileName>("parameter_file_name")),
+    _parameters(declareValueByName<std::vector<Real>>("parameter", REPORTER_MODE_REPLICATED)),
+    _grid(declareValueByName<std::vector<std::vector<Real>>>("gird", REPORTER_MODE_REPLICATED)),
+    _axes(declareValueByName<std::vector<int>>("axes", REPORTER_MODE_REPLICATED)),
+    _step(declareValueByName<std::vector<unsigned int>>("step", REPORTER_MODE_REPLICATED)),
+    _dim(declareValueByName<unsigned int>("dim", REPORTER_MODE_REPLICATED))
+{
+  GriddedData::parse(_dim, _axes, _grid, _parameters, _step, _file_name);
```

Comment:
Does the parse function do all of the appropriate checking for size, etc of the gridded data input?

Reply:
it checks the size, the consistency of the size of axes with the provided data, checks if the dim=0 and if it can not find/open the file.